text/0061-personal-storage.md

```diff
@@ -0,0 +1,228 @@
+The Main Identity is the first derived key of the HD wallet, therefore, every subsequent derived key (POD) is the _subentity_ of the user.
+
+There is possibility to create a _shared POD_ which is a derived key from one of the parties.
+The public key of the shared POD can be shared along with the key used for encryption with another user in order to read data.
+Optionally, the file structure of a shared POD can be merged with another POD for listing directories or files, thereby, mimicking a multi-user file system.
+
+![pods](../resources/pods.png)
+
+The versioning is achieved by using Epoch-based Swarm Feeds that allows sporadic updates under a topic.
+The updates are achored to the time so one can find the nearest update of a given point in time.
+In the feed payloads, all related metadata can be found for files and folders such as its type, name, creation data, etc.
+_For more information about Epoch-based feeds, read the Book of Swarm (1.0) at 4.3.4 chapter._
+
+# Reference-level explanation
+The reference implementation of the Personal Storage can be found in the [fdp-storage](https://github.com/fairDataSociety/fdp-storage) JS library.
+
+## Main Identity
+The Main Identity is the first derived address in the HD wallet. 
+Its derivation path is `m/44'/60'/0'/0/0` but _optionally_ clients can support multiple account handling with incrementing the 3rd `Account` parameter such as `m/44'/60'/1'/0/0` for the second account derived from the seed.
+
+This address is used to make blockchain transactions and manage PODs meanwhile other addresses (PODs) only do off-chain operations for example Swarm Feed signing for dApps.
+
+## POD
+All POD have a `name` property in order to refer that with a human-readable string.
+Each POD type has additional attributes required to initialize the drive in question.
+
+The private POD is a derived keypair from the HD Wallet on path `m/44'/60'/0'/0/${index}` where `${index}` starts from `1` and it is incremented by creating a new POD.
+This index has to be stored along with a 32 bytes of encryption key called `password` with which all metadata will be encrypted in a POD.
+```ts
+interface PrivatePod {
+  name: string
+  password: Bytes<32>
+  index: number
+}
+```
+
+One can share their private POD that becames a Shared POD.
+For that, a shared link is created that the other user can insert to their POD list. 
+It contains the before mentioned `password` and the Ethereum Address of the POD.
+Latter is needed to query the Feed Updates of the POD.
+All together, a Shared POD structure looks like the following
+
+Shared POD
+```ts
+interface SharedPod {
+  name: string
+  password: Bytes<32>
+  address: EthAddress
+}
+```
+
+### Serialization
+The serialization of these pods in the `PODs` consensual topic is
+```ts
+interface PodList {
+  pods: { name: string, password: HexString, index: number }[]
+  sharedPods: { name: string, password: HexString, address: HexString }[]
+}
```

Comment:
Do we need this information shared?

```suggestion
interface PodList {
  pods: []
  sharedPods: []
}
```

Reply:
we do not share this information, it is encrypted by the hash of the private key of the main account that you can read below.

we also need `index` in `pods`, since it is required to derive the associated wallet.
the array index cannot be used, because pod items can be removed from the array.

Reply:
okay, I get it, But we are comparing different things here then. 
The pod list api on fairOS side is client oriented, the clients do not require all this info from fairOS. 

The REST api do not need this. The internal api must have this info for wallet derivation but the client facing api does not need all these, because
- clients do not have an exposed api to get the private key for wallet derivation
- clients do not encrypt or decrypt content, encryption and decryption is done internally with podPassword

Reply:
this whole document just describes the data-structure that must be used in Swarm to integrate to the personal storage concept, it does not introduce any API requirements.

Reply:
The underlying data structure is same. We are good then.